File: 백준/Silver/11399. ATM/ATM.cc
#include<iostream>
#include<algorithm>
#include <vector>
using namespace std;
int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int sum = 0;
	int M;
	cin >> M;
	vector<int> A(M, 0);


	for (int i = 0; i < M; i++) {
		cin >> A[i];
	}

	sort(A.begin(), A.end());

	int time = 0, tmp = 0;
	for (int i = 0; i < A.size(); i++) {
		tmp += A[i];
		time += tmp;
	}
	cout << time << "\n";
	
}

File: 백준/Silver/11399. ATM/ATM.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		
		int A[] = new int[N];
		
		String[] input = br.readLine().split(" ");
		
		for(int i = 0;i < N;i++) {
			A[i] = Integer.parseInt(input[i]);
		}
		
		int sum = 0;
		
		Arrays.sort(A);
		
		int time = 0, temp = 0;
		
		for(int i = 0;i < N;i++) {
			temp += A[i];
			time += temp;
		}
		
		System.out.println(time);
	}

}
